### DISCUSSION TITLE
Is it possible to validate one JSON schema structure with another JSON schema with Java?

### DISCUSSION THREAD

**Discussion by abhikt48** [json-schema-org/community#156] [Q&A] [answered]
I need to validate JSON schema structure with referenced JSON schema. I need to make sure that JSON schema adhere reference structure.

Example - Below JSON schema uses reference of **"$schema": "https://json-schema.org/draft/2020-12/schema"** . I need to make sure that below Json schema is valid against **"$schema": "https://json-schema.org/draft/2020-12/schema"**

**Is It possible to validate one JSON schema structure with another JSON schema ?**

I haven't found any Java SDK/library yet to validate one schema against another json schema. It will be great if anyone can provide any input to validate Json schema against another json schema

   ```
 {
      "$schema": "https://json-schema.org/draft/2020-12/schema",
      "$id": "https://test.com/employee.schema.json",
      "title": "employee",
      "type": "object",
      "properties": {
        "firstName": {
          "type": "string",
          "description": "The employee's first name."
        },
        "lastName": {
          "type": "string",
          "description": "The employee's last name."
        },
        "age": {
          "description": "Age",
          "type": "integer",
          "minimum": 0
        }
      }
    }
```

**Comment by Relequestual** [comment, marked as answer]
A JSON Schema is "just JSON", so treat the schema like JSON and do validation as you usually would.
You should download the meta-schema, although this can be updated without notice or versioning (unlikely, but possible).

**Reply by abhikt48** [reply]
Thanks for details. Really helps.

**Reply by rpgoldman** [reply]
> A JSON Schema is "just JSON", so treat the schema like JSON and do validation as you usually would. You should download the meta-schema, although this can be updated without notice or versioning (unlikely, but possible).

I was wondering if you could describe how to download the meta-schema. When I look on the json-schema page, I don't see a single-file version of the meta-schema.  I'm using `jsonschema` which does not (AFAICT) dereference schema URLs.

Is there a single-file version available? If not, could one assemble one, or would a standard evaluator be able to assemble one out of a directory structure?  There is this question on Stack Overflow, but it was not answered: https://stackoverflow.com/questions/69452886/how-to-validate-json-schema-against-metaschema-in-linux-cli

Sorry if these are dumb questions but ... keyword search _really_ isn't good for finding out how to validate against the meta-schema!  Thank you

**Reply by jdesrosiers** [reply]
This [gist](https://gist.github.com/jdesrosiers/2c7fc711854b1bb3ab8c7c6f19438d7a) should be helpful. I included the bundled meta-schema and the script used to bundle it if you need to generate a fresh copy.

**Comment by karenetheridge** [comment]
I already answered this for you yesterday: https://stackoverflow.com/questions/71326806/how-to-validate-json-schema-structure-with-another-json-schema-with-java/71328611#71328611

**Reply by abhikt48** [reply]
Thank you so much for quick response. Somehow, I didn’t understand completely with yesterday response. With today SDK details and above response cleared my doubt. Thanks again for quick help.
